src/vscode-dts/vscode.d.ts

```diff
@@ -19654,6 +19737,306 @@ declare module 'vscode' {
 		 */
 		canSendRequest(chat: LanguageModelChat): boolean | undefined;
 	}
+
+	/**
+	 * A tool that is available to the language model via {@link LanguageModelChatRequestOptions}. A language model uses all the
+	 * properties of this interface to decide which tool to call, and how to call it.
+	 */
+	export interface LanguageModelChatTool {
+		/**
+		 * The name of the tool.
+		 */
+		name: string;
+
+		/**
+		 * The description of the tool.
+		 */
+		description: string;
+
+		/**
+		 * A JSON schema for the parameters this tool accepts.
+		 */
+		parametersSchema?: object;
+	}
+
+	/**
+	 * A tool-calling mode for the language model to use.
+	 */
+	export enum LanguageModelChatToolMode {
+		/**
+		 * The language model can choose to call a tool or generate a message. Is the default.
+		 */
+		Auto = 1,
+
+		/**
+		 * The language model must call one of the provided tools. Note- some models only support a single tool when using this
```

Comment:
Azure OpenAI doesn't fully support this required mode. Is this comment sufficient or would it be better to go back to `toolChoice?: string`

Reply:
I think that's fair. The extension providing/implementing LMs should be in charge of that and document "how" they implement the API

Reply:
Where do you think vscode-copilot should document that?